src/app/create-template/main-server-action.ts

```diff
@@ -103,5 +98,17 @@ export const mainServerAction = async (
     };
   }
 
-  return serverAction(formState, formData);
+  const serverActionResult = serverAction(formState, formData);
+  if (serverActionResult.validationError) {
+    return serverActionResult;
+  }
+
+  const session: Session = removeUndefinedFromObject({
+    nhsAppTemplateName: serverActionResult.nhsAppTemplateName,
```

Comment:
Lets include the initial part of the form as well, i.e. the template type selection.